packages/admin/admin/src/form/file/FileSelect.tsx

```diff
@@ -0,0 +1,263 @@
+            {value.length > 0 && (
+                <FileList {...slotProps?.fileList}>
+                    {value.map((file, index) => {
+                        return (
+                            <FileListItem
+                                key={index}
+                                name={file.name}
+                                size={"size" in file ? file.size : undefined}
+                                loading={"loading" in file && Boolean(file.name)}
+                                skeleton={"loading" in file && !file.name}
+                                error={"error" in file && file.error}
+                                onClickDownload={
+                                    "error" in file || !onDownload
+                                        ? undefined
+                                        : () => {
+                                              onDownload(file);
+                                          }
+                                }
+                                onClickDelete={() => onRemove(file)}
+                                {...slotProps?.fileListItem}
+                            />
+                        );
+                    })}
+                </FileList>
+            )}
+            {Boolean(error) && (
+                <Error {...slotProps?.error}>
+                    {errorIcon}
+                    <ErrorMessage variant="caption" color="error" {...slotProps?.errorMessage}>
+                        {error}
+                    </ErrorMessage>
+                </Error>
+            )}
+            <FilesInfoText {...slotProps?.filesInfoText}>
+                {typeof maxFiles === "number" ? (
+                    maxFiles === 1 ? (
+                        <FormattedMessage
+                            id="comet.fileSelect.singleFileMaximumSize"
+                            defaultMessage="Maximum 1 file, {maxFileSize}"
+                            values={{
+                                maxFileSize: <PrettyBytes value={maxFileSize} />,
+                            }}
+                        />
+                    ) : (
+                        <FormattedMessage
+                            id="comet.fileSelect.maximumFileNumberAndSize"
+                            defaultMessage="Maximum {maxFiles, plural, one {# file} other {# files}}, {maxFileSize} per file."
+                            values={{
+                                maxFileSize: <PrettyBytes value={maxFileSize} />,
+                                maxFiles,
+                            }}
+                        />
+                    )
+                ) : (
+                    <FormattedMessage
+                        id="comet.fileSelect.maximumSizePerFile"
+                        defaultMessage="Maximum {maxFileSize} per file."
+                        values={{
+                            maxFileSize: <PrettyBytes value={maxFileSize} />,
+                        }}
+                    />
```

Comment:
I think this would be a rare use-case but probably a good idea. 
I'd keep the default value though and allow setting it to `null`.

Reply:
Shouldn't the default value be something that  a wrapper sets? For instance, public uploads have a maximum file size. It would make sense to set the max file size there.

Reply:
Yes, your probably right 👍🏼